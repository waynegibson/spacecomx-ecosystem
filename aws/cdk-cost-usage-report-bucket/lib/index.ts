export * from './stacks/cost-usage-report-bucket-stack'
