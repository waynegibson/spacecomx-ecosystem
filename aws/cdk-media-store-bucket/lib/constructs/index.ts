export * from './cdn-provider-access-policy'
