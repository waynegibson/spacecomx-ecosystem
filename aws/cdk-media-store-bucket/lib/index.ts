export * from './stacks/media-store-bucket-stack'
