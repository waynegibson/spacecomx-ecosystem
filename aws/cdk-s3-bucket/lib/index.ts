export * from './s3-bucket'
