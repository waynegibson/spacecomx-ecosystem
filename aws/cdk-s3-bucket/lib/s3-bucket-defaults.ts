import { RemovalPolicy } from 'aws-cdk-lib'
import * as s3 from 'aws-cdk-lib/aws-s3'

/**
 * @internal
 */
export function DefaultS3Props (loggingBucket?: s3.Bucket, lifecycleRules?: s3.LifecycleRule[]): s3.BucketProps {
  return {
    versioned: true,
    enforceSSL: true,
    encryption: s3.BucketEncryption.S3_MANAGED,
    blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
    autoDeleteObjects: false,
    removalPolicy: RemovalPolicy.RETAIN,
    lifecycleRules: lifecycleRules ?? [],
    serverAccessLogsBucket: loggingBucket
  }
}
