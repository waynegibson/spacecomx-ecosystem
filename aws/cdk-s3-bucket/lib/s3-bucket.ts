import { Construct } from 'constructs'
import type * as s3 from 'aws-cdk-lib/aws-s3'
import { Bucket } from 'aws-cdk-lib/aws-s3'
import { consolidateProps } from '@spacecomx/cdk-utils'
import { DefaultS3Props } from './s3-bucket-defaults'

export interface S3BucketProps {
  /**
   * User provided props to override the default props for the S3 Bucket.
   *
   * @default - Default props are used
   */
  readonly bucketProps?: s3.BucketProps
}

export class S3Bucket extends Construct {
  public readonly s3Bucket: s3.Bucket

  constructor(scope: Construct, id: string, props: S3BucketProps = {}) {
    super(scope, id)

    const combinedBucketProps = consolidateProps(DefaultS3Props(), props.bucketProps)
    const s3BucketResponse = new Bucket(scope, 'S3Bucket', combinedBucketProps)

    this.s3Bucket = s3BucketResponse
  }
}
