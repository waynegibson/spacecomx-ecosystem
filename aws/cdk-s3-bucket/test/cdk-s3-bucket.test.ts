/* eslint-disable no-new */
import { Stack } from 'aws-cdk-lib'
import { Template } from 'aws-cdk-lib/assertions'
import { S3Bucket } from '../lib'

describe('Default S3 Bucket construct -', () => {
  test('it should create a default s3 bucket.', () => {
    const stack = new Stack()
    
    new S3Bucket(stack, 'test-s3-bucket', {})

    const t = Template.fromStack(stack)

    expect(t).toMatchSnapshot()

    t.hasResourceProperties('AWS::S3::Bucket', {
      BucketEncryption: {
        ServerSideEncryptionConfiguration: [
          {
            ServerSideEncryptionByDefault: {
              SSEAlgorithm: 'AES256',
            },
          },
        ],
      },
      PublicAccessBlockConfiguration: {
        BlockPublicAcls: true,
        BlockPublicPolicy: true,
        IgnorePublicAcls: true,
        RestrictPublicBuckets: true,
      },
    })

    t.hasResource('AWS::S3::Bucket', {
      DeletionPolicy: 'Retain',
      UpdateReplacePolicy: 'Retain',
    })
  })
})
