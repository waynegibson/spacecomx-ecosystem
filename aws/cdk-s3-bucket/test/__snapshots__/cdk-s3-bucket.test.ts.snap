// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default S3 Bucket construct - it should create a default s3 bucket. 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "S3Bucket07682993": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
        "VersioningConfiguration": {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "S3BucketPolicyF560589A": {
      "Properties": {
        "Bucket": {
          "Ref": "S3Bucket07682993",
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": "false",
                },
              },
              "Effect": "Deny",
              "Principal": {
                "AWS": "*",
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "S3Bucket07682993",
                    "Arn",
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "S3Bucket07682993",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
